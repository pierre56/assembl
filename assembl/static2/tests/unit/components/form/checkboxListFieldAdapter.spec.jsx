// @flow
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';

import DummyForm from './dummyForm';
import CheckboxListFieldAdapter from '../../../../js/app/components/form/checkboxListFieldAdapter';

configure({ adapter: new Adapter() });

describe('CheckboxListFieldAdapter component', () => {
  const onBlurSpy = jest.fn();
  const onChangeSpy = jest.fn();
  const onFocusSpy = jest.fn();
  const props = {
    input: {
      name: 'languages',
      onBlur: onBlurSpy,
      onChange: onChangeSpy,
      onFocus: onFocusSpy,
      value: [
        {
          isChecked: false,
          label: 'German',
          value: 'de'
        },
        {
          isChecked: false,
          label: 'English',
          value: 'en'
        },
        {
          isChecked: true,
          label: 'French',
          value: 'fr'
        }
      ]
    },
    meta: {
      error: '',
      touched: false
    }
  };

  it('should toggle isChecked when a checkbox is changed', () => {
    const wrapper = shallow(<CheckboxListFieldAdapter {...props} />);
    wrapper
      .find('Checkbox[value="fr"]')
      .props()
      .onChange();
    const expected = [
      {
        isChecked: false,
        label: 'German',
        value: 'de'
      },
      {
        isChecked: false,
        label: 'English',
        value: 'en'
      },
      {
        isChecked: false,
        label: 'French',
        value: 'fr'
      }
    ];
    expect(onChangeSpy).toHaveBeenCalledWith(expected);
  });

  it('should render a list of checkboxes multivalued field', () => {
    const component = renderer.create(
      <DummyForm>
        <CheckboxListFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a list of checkboxes multivalued field without value', () => {
    props.input.value = undefined;
    const component = renderer.create(
      <DummyForm>
        <CheckboxListFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a list of checkboxes multivalued field untouched with error', () => {
    props.input.value = undefined;
    props.meta = {
      error: 'This field is required',
      touched: false
    };
    const component = renderer.create(
      <DummyForm>
        <CheckboxListFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a list of checkboxes multivalued field touched with error', () => {
    props.meta = {
      error: 'This field is required',
      touched: true
    };
    const component = renderer.create(
      <DummyForm>
        <CheckboxListFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});