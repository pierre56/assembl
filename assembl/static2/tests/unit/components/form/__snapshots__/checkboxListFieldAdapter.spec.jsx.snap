// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxListFieldAdapter component should render a list of checkboxes multivalued field 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="checkbox"
    >
      <label
        title="German"
      >
        <input
          checked={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="de"
        />
        German
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        title="English"
      >
        <input
          checked={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="en"
        />
        English
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        title="French"
      >
        <input
          checked={true}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="fr"
        />
        French
      </label>
    </div>
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`CheckboxListFieldAdapter component should render a list of checkboxes multivalued field touched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group has-error"
  />
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`CheckboxListFieldAdapter component should render a list of checkboxes multivalued field untouched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  />
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`CheckboxListFieldAdapter component should render a list of checkboxes multivalued field without value 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  />
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
